aptos-move/e2e-move-tests/src/tests/aggregator_v2.data/pack/sources/aggregator_v2_test.move

```diff
@@ -51,120 +52,171 @@ module 0x1::aggregator_v2_test {
+        let value = option::borrow_mut(option_data);
+        f(value)
+    }
+
 
     public entry fun new(account: &signer, i: u64, limit: u128) acquires AggregatorStore {
-        let addr = signer::address_of(account);
-        let aggregators = &mut borrow_global_mut<AggregatorStore>(addr).aggregators_u128;
-        let aggregator = aggregator_v2::create_aggregator(limit);
-        table::add(aggregators, i, aggregator);
+        insert<Aggregator<u128>>(account, i, aggregator_v2::create_aggregator(limit));
+    }
+
+    public fun call_read(account: &signer, i: u64): u128 acquires AggregatorStore {
+        for_element_ref<Aggregator<u128>, u128>(account, i, |aggregator| aggregator_v2::read(aggregator))
     }
 
     public entry fun try_add(account: &signer, i: u64, value: u128) acquires AggregatorStore {
-        let addr = signer::address_of(account);
-        let aggregators = &mut borrow_global_mut<AggregatorStore>(addr).aggregators_u128;
-        let aggregator = table::borrow_mut(aggregators, i);
-        aggregator_v2::try_add(aggregator, value);
+        for_element_mut<Aggregator<u128>, bool>(account, i, |aggregator| aggregator_v2::try_add(aggregator, value));
+    }
+
+    public entry fun add(account: &signer, i: u64, value: u128) acquires AggregatorStore {
+        for_element_mut<Aggregator<u128>, bool>(account, i, |aggregator| {
+            aggregator_v2::add(aggregator, value);
+            true
+        });
     }
 
     public entry fun try_sub(account: &signer, i: u64, value: u128) acquires AggregatorStore {
-        let addr = signer::address_of(account);
-        let aggregators = &mut borrow_global_mut<AggregatorStore>(addr).aggregators_u128;
-        let aggregator = table::borrow_mut(aggregators, i);
-        aggregator_v2::try_sub(aggregator, value);
+        for_element_mut<Aggregator<u128>, bool>(account, i, |aggregator| aggregator_v2::try_sub(aggregator, value));
     }
 
-    public entry fun try_sub_add(account: &signer, i: u64, a: u128, b: u128) acquires AggregatorStore {
-        let addr = signer::address_of(account);
-        let aggregators = &mut borrow_global_mut<AggregatorStore>(addr).aggregators_u128;
-        let aggregator = table::borrow_mut(aggregators, i);
-        aggregator_v2::try_sub(aggregator, a);
-        aggregator_v2::try_add(aggregator, b);
+    public entry fun sub(account: &signer, i: u64, value: u128) acquires AggregatorStore {
+        for_element_mut<Aggregator<u128>, bool>(account, i, |aggregator| {
+            aggregator_v2::sub(aggregator, value);
+            true
+        });
     }
 
     public entry fun materialize(account: &signer, i: u64) acquires AggregatorStore {
-        let addr = signer::address_of(account);
-        let aggregators = &borrow_global<AggregatorStore>(addr).aggregators_u128;
-        let aggregator = table::borrow(aggregators, i);
-        aggregator_v2::read(aggregator);
+        call_read(account, i);
```

Comment:
If materialize and call_read are the same, why have two functions?

Reply:
entry function cannot return result, it needs to be void